From f38853b916048b4f5496b23ea12df1d6b8762d06 Mon Sep 17 00:00:00 2001
From: Simon Wunderlich <sw@simonwunderlich.de>
Date: Fri, 9 Feb 2018 15:32:58 +0100
Subject: [PATCH 2/3] net-snmp: remove cached user credentials

net-snmp caches the user credentials in /usr/lib/snmp/*. If the user
credentials are changed later in UCI, these changes will not get updated
upon a restart and net-snmp will run with the old credentials.

Therefore, clean up the cache on each service start/stop.

Signed-off-by: Simon Wunderlich <sw@simonwunderlich.de>
---
 net/net-snmp/files/snmpd.init | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/net-snmp/files/snmpd.init b/net/net-snmp/files/snmpd.init
index 9ebcb491c..2064d595a 100644
--- a/net/net-snmp/files/snmpd.init
+++ b/net/net-snmp/files/snmpd.init
@@ -283,6 +283,7 @@ snmpd_setup_fw_rules() {
 
 start_service() {
 	[ -f "$CONFIGFILE" ] && rm -f "$CONFIGFILE"
+	rm -rf /usr/lib/snmp/*
 
 	config_load snmpd
 
@@ -338,6 +339,7 @@ start_service() {
 
 stop_service() {
 	[ -f "$CONFIGFILE" ] && rm -f "$CONFIGFILE"
+	rm -rf /usr/lib/snmp/*
 	procd_set_config_changed firewall
 }
 
-- 
2.19.1

